modules/data/event-hub/main.bicep

```diff
@@ -0,0 +1,245 @@
+@description('Optional. Name for the Event Hub cluster, Alphanumerics and hyphens characters, Start with letter, End with letter or number.')
+param clusterName string = 'null'
+
+@description('Optional. The quantity of Event Hubs Cluster Capacity Units contained in this cluster.')
+param clusterCapacity int = 1
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Tags of the resource.')
+param tags object = {}
+
+@description('''Optional. The name of the event hub namespace to be created.
+Below paramters you can pass while creating the Azure Event Hub namespace.
+sku: (Optional) Possible values are "Basic" or "Standard" or "Premium". Detault to 'Standard'.
+capacity: (Optional) int, The Event Hubs throughput units for Basic or Standard tiers, where value should be 0 to 20 throughput units.The Event Hubs premium units for Premium tier, where value should be 0 to 10 premium units. Default to 1.
+zoneRedundant: (Optional) bool, Enabling this property creates a Standard Event Hubs Namespace in regions supported availability zones. Default to false.
+isAutoInflateEnabled: (Optional) bool,  Value that indicates whether AutoInflate is enabled for eventhub namespace. Only available for "Standard" sku. Default to false.
+maximumThroughputUnits: (Optional) int, Upper limit of throughput units when AutoInflate is enabled, value should be within 0 to 20 throughput units. ( '0' if AutoInflateEnabled = true)
+disableLocalAuth: (Optional) bool, This property disables SAS authentication for the Event Hubs namespace. Default to false.
+kafkaEnabled: (Optional) bool, Value that indicates whether Kafka is enabled for eventhub namespace. Default to true.
+''')
+param eventHubNamespaces object = {}
+
+@description(''' Optional. Name for the eventhub to be created.
+Below paramters you can pass while the creating Azure Event Hub
+messageRetentionInDays: (Optional) int, Number of days to retain the events for this Event Hub, value should be 1 to 7 days. Default to 1.
+partitionCount: (Optional) int, Number of partitions created for the Event Hub. Default to 2
+eventHubNamespaceName: (Optional) string, Name of the Azure Event Hub Namespace
+status: (Optional) string, Enumerates the possible values for the status of the Event Hub. 'Active','Creating','Deleting','Disabled','ReceiveDisabled','Renaming','Restoring','SendDisabled','Unknown'. Default to 'Active'
+''')
+param eventHubs object = {}
```

Comment:
does this need a default value for a minimum deployment?

Reply:
we need to pass namespace name here, So I don't think we can do it with default value.